### PR TITLE
Issue2080 Accept ttf-file-extension for mimetype font/sfnt

### PR DESCRIPTION
**Pull request by r-peschke**

### CODE REVIEW

**Review by jsangmeister — CHANGES_REQUESTED** [review]
Please add some comments on what is done and why as the procedure for mimetype checking is becoming increasingly complex and difficult to understand.

File: openslides_backend/action/actions/mediafile/upload.py [hunk 1]
```diff
@@ -112,10 +112,21 @@ def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
...
-            mismatched = not any(
-                [filename_.endswith(extension) for extension in possible_extensions]
-            )
+            mismatched = check_extension(filename_, possible_extensions)
+            if mismatched:
+                possible_extensions = (
+                    python_magic.Magic(extension=True)  # type: ignore
+                    .from_buffer(decoded_file)
+                    .split("/")
+                )
```
Comment (luisa-beerboom) [inline openslides_backend/action/actions/mediafile/upload.py:128]: What does this code do?

**Review by luisa-beerboom — COMMENTED** [review]
I just noticed in ln 102 of `upload.py` it checks a mimetype against "application.json". Wouldn't that have to be "application/json" to have any effect?

**Comment by r-peschke** [comment]
> I just noticed in ln 102 of `upload.py` it checks a mimetype against "application.json". Wouldn't that have to be "application/json" to have any effect?

Yes you are right!
This case now is resolved with the pygments.lexers

**Comment by r-peschke** [comment]
> Please add some comments on what is done and why as the procedure for mimetype checking is becoming increasingly complex and difficult to understand.

I think I guess what you are asking for. Let's talk about tomorrow!

**Review by jsangmeister — APPROVED** [review]

**Pull request merged**